### ISSUE TITLE
Receiving ERROR: Could not get HiyaCFW running HiyaCFW_Helper.py Python 3.7.2

### ISSUE THREAD

**Issue by Gurney-Halleck** [mondul/HiyaCFW-Helper#17]

**Comment by Gurney-Halleck** [comment]
This is exact copy of the error message:

Checking NAND file...
- eMMC CID: 5DF41310D7034D303046504100001500
- Console ID: 0820202602103137

Downloading latest HiyaCFW release...
- Extracting HiyaCFW archive...
ERROR: Could not get HiyaCFW

**Comment by Gurney-Halleck** [comment]
I am running Mac OS X 10.9.5 with Python 3.7.2 installed running the HiyaCFW_Helper.py 3.0.1 script.

**Comment by Gurney-Halleck** [comment]
I have disabled firewall and have no active anti-virus or security software running while I am running the helper from Terminal.

**Comment by mondul** [comment]
If you got to the extracting part seems to be a problem with 7-Zip then. Please ```cd``` to the helper's folder, run ```Darwin/7za``` and tell me if you get any output.

**Comment by Palmsay** [comment]
I have the exact same issue. Regardless of what machine I run the helper program on, it keeps returning "ERROR: Could not get HiyaCFW". Perhaps I could be given a link to a direct download of the CFW and be given instructions on where to put said files on the SD?

**Comment by mondul** [comment]
You need your NAND dump in order to generate the CFW for your device. If you want you can upload your NAND to Google Drive, share the link with me and I can generate the CFW files for you.

**Comment by Palmsay** [comment]
I have my NAND dumped. However, I cannot use your HiyaCFW-Helper tool to successfully download the HiyaCFW from the internet. As I've said, the program returns "ERROR: Could not get HiyaCFW". I know the Python attempts to download HiyaCFW.7z and several other files. Is there a possibility to download these required files manually and point the script to their location without having to rely on an internet connection for the helper program?

**Comment by mondul** [comment]
I will restore that feature for HiyaCFW then but not for TWiLightMenu.

**Comment by mondul** [comment]
@Gurney-Halleck @Palmsay Uploaded a new release, please test.

**Comment by Palmsay** [comment]
This update worked and HiyaCFW was able to be installed successfully. However, there still seems to be an issue with your code in how it pulls the HiyaCFW.7z and Twilight Menu 7z file, as well. However, I was able to circumvent this by disabling the optional Twilight Menu, manually providing the HiyaCFW.7z  and then manually installing Twilight Menu. Perhaps change your script to REQUIRE the user to provide the 7z archives opposed to pulling them from the web. Releases are few and far between, typically, and I personally think you’d be better off having the user provide the files.

**Comment by mondul** [comment]
I would like to see the real error that happens when fails, that's why this latest release prints the error object to the console. I've tested the helper on 7 environments (my mac, windows 7 32-bit, windows 10 64-bit, debian stretch 32-bit, ubuntu 18.04 64-bit, Fedora 29 64-bit and manjaro 64-bit) and have not been able to replicate this error. Please someone attach a console log with the error.

**Comment by cheeseslope** [comment]
I was also getting "ERROR: Could not get HiyaCFW", but I found a workaround.

I was running the Helper from a folder on my Desktop. I noticed that the script _was_ downloading the HiyaCFW.7z, but it was downloading to my root user folder (/Users/[username]). So I moved the Helper files directly to that folder and ran it again successfully.

_I’m running Mac OS X 10.14.3 with Python 3.7.3 installed running the HiyaCFW_Helper.py 3.0.2 script._
